### PR TITLE
doc: release-notes-3.7: Beagle notes

### PR DESCRIPTION
**Pull request by jadonk**
Add notes on updates to support for BeagleBoard.org boards.

### CODE REVIEW

**Comment by jadonk** [comment]
@Ayush1325 please confirm I didn't miss anything.

**Review by nashif — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -379,6 +379,9 @@ Boards & SoC Support
   * nRF54H20 PDK (pre-release) converted to :ref:`nrf54h20dk_nrf54h20`
   * PPR core target in :ref:`nrf54h20dk_nrf54h20` runs from RAM by default. A
     new ``xip`` variant has been introduced which runs from MRAM (XIP).
+  * Refactored :ref:`beagleconnect_freedom` external antenna switch handling.
+  * Fix zephyr:code-sample:`pwm-blinky` code sample to demonstrate PWM support for
+    :ref:`beagleconnect_freedom`.
```
Comment (Ayush1325) [inline doc/releases/release-notes-3.7.rst:384]: I think the following also need to be added:
- gpio-i2c-switch: https://github.com/zephyrproject-rtos/zephyr/pull/64693
- subg: https://github.com/zephyrproject-rtos/zephyr/pull/65048

Not sure if the 2nd one is part of beagle notes though

**Review by kartben — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst (hunk 1 above)
Comment (jadonk) [reply doc/releases/release-notes-3.7.rst:384]: Do not see those in `git log v3.6.0..main`. Perhaps they went into 3.6?

**Comment by jadonk** [comment]
I fixed a typo in my commit title and moved the PWM update to the PWM section.

**Review by Ayush1325 — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**